libs/blocks/global-navigation/blocks/profile/profile.js

```diff
@@ -9,87 +10,89 @@ function decorateEmail(email) {
-  const accountText = blockEl.querySelector('div > div > p:nth-child(2) a').innerHTML;
-  const profileViewAccount = createTag('p', { class: 'gnav-profile-account' }, accountText);
-  profileDetails.append(profileName, profileEmail, profileViewAccount);
+const decorateAction = (actionEl) => {
+  if (!actionEl) return '';
+  actionEl.href = decorateProfileLink(actionEl.href, 'adminconsole');
+  return toFragment`<li class="gnav-profile-action">${actionEl}</li>`;
+};
 
-  if (sections.manage.items.team?.id) {
-    const teamLink = blockEl.querySelector('div > div > p:nth-child(3) a');
-    teamLink.href = decorateProfileLink(teamLink.href, 'adminconsole');
-    const manageTeam = createTag('li', { class: 'gnav-profile-action' }, teamLink);
-    profileActions.append(manageTeam);
+class Profile {
+  constructor({
+    decoratedEl,
+    avatarImgEl,
+    sections,
+    toggleMenu,
+    profileButtonEl,
+    accountLinkEl,
+    signOutEl,
+    manageTeamsEl,
+    manageEnterpriseEl,
+  }) {
+    this.sections = sections;
+    this.avatarImgEl = avatarImgEl;
+    this.accountLinkEl = accountLinkEl;
+    this.signOutEl = signOutEl;
+    this.manageTeamsEl = manageTeamsEl;
+    this.manageEnterpriseEl = manageEnterpriseEl;
+    this.toggleMenu = toggleMenu;
+    this.profileButtonEl = profileButtonEl;
+    this.decoratedEl = decoratedEl;
+    this.init();
   }
 
-  if (sections.manage.items.enterprise?.id) {
-    const manageLink = blockEl.querySelector('div > div > p:nth-child(4) a');
-    manageLink.href = decorateProfileLink(manageLink.href, 'adminconsole');
-    const manageEnt = createTag('li', { class: 'gnav-profile-action' }, manageLink);
-    profileActions.append(manageEnt);
+  async init() {
+    const { displayName, email } = await window.adobeIMS.getProfile();
+    if (this.profileButtonEl) this.profileButtonEl.setAttribute('aria-label', displayName);
+    this.displayName = displayName;
+    this.email = email;
+    this.decoratedEl.append(this.menu());
+    this.decoratedEl.addEventListener('click', () => this.toggleMenu(this.decoratedEl));
+    this.decoratedEl.dispatchEvent(new Event('profile_ready'));
   }
 
-  const signOutLink = blockEl.querySelector('div > div > p:nth-child(5) a');
-  signOutLink.addEventListener('click', (e) => {
-    e.preventDefault();
-    window.adobeIMS.signOut();
-  });
-  const signOut = createTag('li', { class: 'gnav-profile-action' }, signOutLink);
-  profileActions.append(signOut);
-
-  profileHeader.append(profileImg, profileDetails);
-  profileMenu.append(profileHeader, profileActions);
-  profileEl.append(profileButton, profileMenu);
-}
-
-async function profile(blockEl, profileEl, toggle, ioResp) {
-  const gnav = profileEl.closest('nav.gnav');
-  gnav.classList.add('signed-in');
+  decorateSignOut() {
+    const signOutLink = toFragment`<li class="gnav-profile-action">${this.signOutEl}</li>`;
+    signOutLink.addEventListener('click', (e) => {
+      e.preventDefault();
```

Comment:
Some recent use-cases brought up the idea of consumers wanting to delay the default IMS sign out behaviour, could you add a TODO comment that we might want to add a way for consumers to execute their own callback before executing the IMS sign out method?